homeassistant/components/icloud.py

```diff
@@ -0,0 +1,956 @@
+            from pyicloud import PyiCloudService
+            from pyicloud.exceptions import PyiCloudFailedLoginException
+            try:
+                # Attempt the login to iCloud
+                self.api = PyiCloudService(self.username,
+                                           self.password,
+                                           cookie_directory=self.cookiedir,
+                                           verify=True)
+            except PyiCloudFailedLoginException as error:
+                _LOGGER.error('Error logging into iCloud Service: %s',
+                              error)
+
+        if self.api is not None:
+            if not self.getevents:
+                self.api.authenticate()
+            for devicename in self.devices:
+                self.devices[devicename].keep_alive()
+            if self.getevents:
+                from_dt = dt_util.now()
+                to_dt = from_dt + timedelta(days=7)
+                events = self.api.calendar.events(from_dt, to_dt)
+                new_events = sorted(events, key=self.get_key)
+                for event in new_events:
+                    tzone = event['tz']
+                    if tzone is None:
+                        tzone = pytz.utc
+                    else:
+                        tzone = timezone(tzone)
+                    tempnow = dt_util.now(tzone)
+                    guid = event['guid']
+                    starttime = event['startDate']
+                    startdate = datetime(starttime[1], starttime[2],
+                                         starttime[3], starttime[4],
+                                         starttime[5], 0, 0, tzone)
+                    endtime = event['endDate']
+                    enddate = datetime(endtime[1], endtime[2], endtime[3],
+                                       endtime[4], endtime[5], 0, 0, tzone)
+                    duration = event['duration']
+                    title = event['title']
+                    location = event['location']
+
+                    strnow = tempnow.strftime("%Y%m%d%H%M%S")
+                    strstart = startdate.strftime("%Y%m%d%H%M%S")
+                    strend = enddate.strftime("%Y%m%d%H%M%S")
+
+                    if strnow > strstart and strend > strnow:
+                        if guid not in self.currentevents:
+                            ievent = IEvent(self.hass, self, guid,
+                                            TYPE_CURRENT)
+                            ievent.update_ha_state()
+                            self.currentevents[guid] = ievent
+                        self.currentevents[guid].keep_alive(starttime,
+                                                            endtime,
+                                                            duration,
+                                                            title,
+                                                            tzone,
+                                                            location)
```

Comment:
A bit more (compact) & readable formatting here would be:

```python
self.currentevents[guid].keep_alive(
    starttime, endtime, duration, title, tzone, location)
```
Open the bracket, immediately newline and 4 spaces indented...